GenHub/GenHub/Features/Content/Services/GeneralsOnline/GeneralsOnlineDiscoverer.cs

```
@@ -0,0 +1,283 @@
+using GenHub.Core.Constants;
+using GenHub.Core.Interfaces.Content;
+using GenHub.Core.Models.Enums;
+using GenHub.Core.Models.GeneralsOnline;
+using GenHub.Core.Models.Results;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GenHub.Features.Content.Services.GeneralsOnline;
+
+/// <summary>
+/// Discovers Generals Online releases by querying the CDN API or falling back to mock data.
```

Why does this fall back to mock data? Is this only for testing and will it be removed, or are you aware and guarding against issues with the GeneralsOnline API being offline for extended periods of time?